### PR TITLE
newlib: improve the nano version

### PR DESCRIPTION
**Pull request by Artturin**
apply patch from gentoo because there's no libstdc++.a libsupc++.a or their nano versions

this matches upstream arm more
https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads

-lc_nano is used in nano.specs but 'libc_nano.a' is not installed without these changes

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/misc/newlib/default.nix [hunk 1]
```diff
@@ -45,6 +54,18 @@ stdenv.mkDerivation rec {
 
   dontDisableStatic = true;
 
+  # apply necessary nano changes from https://developer.arm.com/-/media/Files/downloads/gnu/12.2.rel1/manifest/copy_nano_libraries.sh?rev=4c50be6ccb9c4205a5262a3925317073&hash=1375A7B0A1CD0DB9B9EB0D2B574ADF66
+  postInstall = lib.optionalString nanoizeNewlib ''
+    mkdir -p $out/${stdenv.targetPlatform.config}/include/newlib-nano
+    cp $out/${stdenv.targetPlatform.config}/include/newlib.h $out/${stdenv.hostPlatform.config}/include/newlib-nano/
+
+    cd $out/${stdenv.targetPlatform.config}/lib
```
Comment (trofi) [inline pkgs/development/misc/newlib/default.nix:62]: Is it safe to `cd` in a hook and not restore the default directory here?

**Review by trofi — APPROVED** [review]

File: pkgs/development/misc/newlib/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/misc/newlib/default.nix]: changed to a subshell

**Comment by peteut** [comment]
@Artturin thanks for taking care of that!

File: pkgs/development/misc/newlib/default.nix [hunk 2]
```diff
@@ -45,8 +54,22 @@ stdenv.mkDerivation rec {
...
+        cp "$f" "''${f%%\.a}_nano.a"
+      done
+    )
+  '';
+
   passthru = {
     incdir = "/${stdenv.targetPlatform.config}/include";
     libdir = "/${stdenv.targetPlatform.config}/lib";
   };
-}
```
Comment (SuperSandro2000) [inline pkgs/development/misc/newlib/default.nix:52]: Where is meta?

Comment (Artturin) [reply pkgs/development/misc/newlib/default.nix:52]: added meta

**Pull request merged**